include/rmm/device_scalar.hpp

```diff
@@ -36,8 +38,25 @@ class device_scalar {
                 "Scalar type must be trivially copyable");
 
   /**
-   * @brief Construct a new `device_scalar`
+   * @brief Construct a new uninitialized `device_scalar`
+   *
+   * @throws `rmm::bad_alloc` if allocating the device memory for
+   *`initial_value` fails
+   * @throws `rmm::cuda_error` if copying `initial_value` to device memory fails
+   *
+   * @param stream Optional, stream on which to perform allocation and copy
+   * @param mr Optional, resource with which to allocate
+   */
+  explicit device_scalar(
+      cudaStream_t stream = 0,
```

Comment:
I think the stream should maybe not be defaulted so the caller is aware of the fact that a stream is involved when constructing, otherwise with default constructions like `device_scalar<int> i{}` it's not obvious that it's storing away a stream. 

Furthermore, given our latest discussions on stream ordered memory, probably need to include some docs about how it's only safe to use on the stream on which it was constructed.

Reply:
Good suggestions.

Reply:
Done. Please review again.